### PR TITLE
[MAINTENANCE] Misc cleanup of GX Cloud helpers

### PR DESCRIPTION
**Pull request by cdkini**
Changes proposed in this pull request:
- `CONTRACT` and `SUITE_VALIDATION_RESULT` are legacy and have been slated for removal for some time.
- Classes should utilize the `GX` prefix
- `get_context()` tests should work when you have env vars set
    - Let's delete them using `monkeypatch` so this is the case

### Definition of Done
- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: tests/data_context/test_get_data_context.py [hunk 1]
```diff
@@ -158,6 +159,10 @@ def test_cloud_context_disabled(set_up_cloud_envs, tmp_path: pathlib.Path):
 def test_cloud_missing_env_throws_exception(
     monkeypatch, empty_ge_cloud_data_context_config
 ):
+    # Delete local env vars (if present)
+    for env_var in GXCloudEnvironmentVariable:
+        monkeypatch.delenv(env_var, raising=False)
+
```
Comment (cdkini) [inline tests/data_context/test_get_data_context.py]: Test fails if you have env vars set - let's ensure we clear out our environment for consistent behavior

File: great_expectations/data_context/cloud_constants.py [hunk 2]
```diff
@@ -5,17 +5,15 @@
...
 
-class GECloudEnvironmentVariable(str, Enum):
+class GXCloudEnvironmentVariable(str, Enum):
     BASE_URL = "GE_CLOUD_BASE_URL"
     ORGANIZATION_ID = "GE_CLOUD_ORGANIZATION_ID"
     ACCESS_TOKEN = "GE_CLOUD_ACCESS_TOKEN"
 
 
-class GeCloudRESTResource(str, Enum):
+class GXCloudRESTResource(str, Enum):
```
Comment (cdkini) [inline great_expectations/data_context/cloud_constants.py:15]: Most of the changes here are from these renames to `GX`

File: great_expectations/data_context/store/ge_cloud_store_backend.py [hunk 3]
```diff
@@ -110,61 +110,48 @@ def get_user_friendly_error_message(
...
-        # Chetan - 20220811 - CONTRACT is deprecated by GX Cloud and is to be removed upon migration of E2E tests
-        GeCloudRESTResource.CONTRACT: "checkpoint_config",
-        GeCloudRESTResource.DATASOURCE: "datasource_config",
-        GeCloudRESTResource.DATA_CONTEXT: "data_context_config",
-        GeCloudRESTResource.DATA_CONTEXT_VARIABLES: "data_context_variables",
-        GeCloudRESTResource.EXPECTATION_SUITE: "suite",
-        GeCloudRESTResource.EXPECTATION_VALIDATION_RESULT: "result",
-        GeCloudRESTResource.PROFILER: "profiler",
-        GeCloudRESTResource.RENDERED_DATA_DOC: "rendered_data_doc",
-        # Chetan - 20220812 - SUITE_VALIDATION_RESULT is deprecated by GX Cloud and is to be removed upon migration of E2E tests
```
Comment (cdkini) [inline great_expectations/data_context/store/ge_cloud_store_backend.py:120]: Removed `CONTRACT` and `SUITE_VALIDATION_RESULT`

File: great_expectations/data_context/store/ge_cloud_store_backend.py [hunk 4]
```diff
@@ -24,8 +25,6 @@
 
 logger = logging.getLogger(__name__)
 
-SUPPORT_EMAIL = "support@greatexpectations.io"
-
```
Comment (cdkini) [inline great_expectations/data_context/store/ge_cloud_store_backend.py:24]: Moved to `cloud_constants.py`

File: great_expectations/data_context/data_context/base_data_context.py [hunk 5]
```diff
@@ -1077,7 +1075,7 @@ def save_profiler(
         key: Union[GeCloudIdentifier, ConfigurationIdentifier]
         if self.ge_cloud_mode:
             key = GeCloudIdentifier(
```
Comment (donaldheppner) [inline great_expectations/data_context/data_context/base_data_context.py]: Can we change the name of GeCloudIdentifier, too?

File: great_expectations/data_context/types/resource_identifiers.py [hunk 6]
```diff
@@ -203,7 +203,7 @@ def from_object(cls, validation_result):
 class GeCloudIdentifier(DataContextKey):
```
Comment (donaldheppner) [inline great_expectations/data_context/types/resource_identifiers.py]: ^^

File: great_expectations/rule_based_profiler/rule_based_profiler.py [hunk 7]
```diff
@@ -1191,7 +1191,7 @@ def get_profiler(
         key: Union[GeCloudIdentifier, ConfigurationIdentifier]
         if ge_cloud_id:
             key = GeCloudIdentifier(
-                resource_type=GeCloudRESTResource.PROFILER, ge_cloud_id=ge_cloud_id
+                resource_type=GXCloudRESTResource.PROFILER, ge_cloud_id=ge_cloud_id
```
Comment (donaldheppner) [inline great_expectations/rule_based_profiler/rule_based_profiler.py:1194]: update the method parameter too? from ge_cloud_id to gx_cloud_id?

File: tests/data_context/store/test_ge_cloud_store_backend.py [hunk 8]
```diff
@@ -282,10 +282,10 @@ def test_list_keys(
 @pytest.mark.unit
 def test_remove_key(
     construct_ge_cloud_store_backend: Callable[
-        [GeCloudRESTResource], GeCloudStoreBackend
+        [GXCloudRESTResource], GeCloudStoreBackend
```
Comment (donaldheppner) [inline tests/data_context/store/test_ge_cloud_store_backend.py:285]: can we also rename the GeCloudStoreBackend?

File: tests/data_context/test_get_data_context.py (hunk 1 above)
Comment (Kilo59) [reply tests/data_context/test_get_data_context.py]: This seems like something we might want to re-use in a fixture.
`clean_cloud_env_vars` / `no_cloud_env` 🤷

Comment (cdkini) [reply tests/data_context/test_get_data_context.py]: yup this was done shortly after!

File: great_expectations/data_context/data_context/base_data_context.py (hunk 5 above)
Comment (cdkini) [reply great_expectations/data_context/data_context/base_data_context.py]: hmm as long as Cloud doesn't directly use this! I don't think they though but lemme peruse the Mercury code first

File: great_expectations/rule_based_profiler/rule_based_profiler.py (hunk 7 above)
Comment (cdkini) [reply great_expectations/rule_based_profiler/rule_based_profiler.py:1194]: sounds good to me!

File: tests/data_context/store/test_ge_cloud_store_backend.py (hunk 8 above)
Comment (cdkini) [reply tests/data_context/store/test_ge_cloud_store_backend.py:285]: that may break existing implementations due to our usage of `instantiate_class_from_config` - do we perhaps want to deprecate that name?

**Review by Kilo59 — APPROVED** [review]

**Pull request merged**